plugins/e57/libE57Format/src/CheckedFile.cpp

```diff
@@ -276,14 +276,14 @@ CheckedFile& CheckedFile::operator<<(const ustring& s)
 
 CheckedFile& CheckedFile::operator<<(int64_t i)
 {
-   stringstream ss;
+   ClassicLocaleStream<stringstream> ss;
```

Comment:
Thanks @hobu I hope I addressed all your points.

> * There's not enough tests that have been added. ClassicLocaleStringStream and friends should have some explicit tests that demonstrate behavior and usage. Also, the place where locale stuff is the most difficult to deal with, Windows, is not tested at all.
> 

I added tests for ClassicLocale stringstreams. About Windows, unfortunately I do not have a computer to test, and neither the knowledge to configure the GitHub actions. @wrenoud tested some things in #3450 using `TEST_SET_LOCALE=nl_NL` in Windows.

> * There's some spots where creating big fat streams all the time are going to hurt performance. I tried to identify a few of those, but there might be some more. Updating them probably means changing PDAL's code a bit more.

I added @rouault's idea of a static thread_local stream to avoid the creation of the stream in `fromString` and a few more places that you pointed as critical.
Most of the changes tried to be minimal to not produce a big PR. But for performance reasons some parts of the code could be optimized as you say. I do not know which parts of the code are called massively, that could impact more the global performance.